aspnetcore/blazor/fundamentals/render-modes.md

```diff
@@ -0,0 +1,295 @@
+
+> [!NOTE]
+> The anatomy of a basic Razor component is fully explained in the <xref:blazor/fundamentals/index> article. Project structure for apps created by a Blazor template are described in the <xref:blazor/project-structure> article. Detailed components coverage is found in the *Components* articles later in the documentation.
+
+### Server render mode
+
+Server render mode results in rendering a component on the server with interactivity over a SignalR connection.
+
+In the following example, the render mode is set to use the Blazor Server hosting model with `@attribute [RenderModeServer]`. The button calls the `UpdateMessage` method when selected. The value of `message` changes, and the component is re-rendered to update the message in the UI.
+
+`RenderMode2.razor`:
+
+```razor
+@page "/render-mode-2"
+@attribute [RenderModeServer]
+
+<button @onclick="UpdateMessage">Click me</button> @message
+
+@code {
+    private string message = "Not clicked yet.";
+
+    private void UpdateMessage()
+    {
+        message = "Somebody clicked me!";
+    }
+}
+```
+
+If using the preceding component locally in a Blazor Web App, place the component in the server-side project's `Components/Pages` folder. The server-side project is the solution's project with a name that doesn't end in `.Client`. When the app is running, navigate to `/render-mode-2` in the browser's address bar.
+
+### Client render mode
+
+Client render mode results in rendering a component on the client with interactivity.
+
+In the following example, the render mode is set to use the Blazor WebAssembly hosting model with `@attribute [RenderModeWebAssembly]`. The button calls the `UpdateMessage` method when selected. The value of `message` changes, and the component is re-rendered to update the message in the UI.
+
+`RenderMode3.razor`:
+
+```razor
+@page "/render-mode-3"
+@attribute [RenderModeWebAssembly]
+
+<button @onclick="UpdateMessage">Click me</button> @message
+
+@code {
+    private string message = "Not clicked yet.";
+
+    private void UpdateMessage()
+    {
+        message = "Somebody clicked me!";
+    }
+}
+```
+
+If using the preceding component locally in a Blazor Web App, place the component in the client-side project's `Pages` folder. The client-side project is the solution's project with a name that ends in `.Client`. When the app is running, navigate to `/render-mode-3` in the browser's address bar.
+
+### Render mode determined at runtime
+
+The render mode can be determined at runtime. The component is initially rendered server-side with interactivity using the Blazor Server hosting model. After the Blazor bundle is downloaded to the client and the .NET client-side runtime activates, the component adopts the Blazor WebAssembly hosting model for client-side rendering and interactivity.
```

Comment:
```suggestion
The Auto render mode determines how to render the component at runtime. The component is initially rendered server-side with interactivity using the Blazor Server hosting model. The Blazor WebAssembly runtime and app bundle is downloaded to the client in the background and cached so that it can be used instead on future visits.
```